src/runtime/virtcontainers/hypervisor.go

```diff
@@ -559,7 +561,7 @@ func (conf *HypervisorConfig) Valid() error {
 
 	if conf.BlockDeviceDriver == "" {
 		conf.BlockDeviceDriver = defaultBlockDriver
-	} else if conf.BlockDeviceDriver == config.VirtioBlock && conf.HypervisorMachineType == "s390-ccw-virtio" {
+	} else if conf.BlockDeviceDriver == config.VirtioBlock && conf.HypervisorMachineType == QemuCCWVirtio {
```

Comment:
Late to the party here but we should ensure that QEMU (or any VMM really) specific code is contained within the VMM implementation itself, and doesn't leak into the (agnostic) hypervisor.go.

More specifically, not every HOST_OS supports QEMU - see https://github.com/Jakob-Naucke/kata-containers/blob/ff17c756d2cb777ce63af2c9aa5a3dd7dc5e2eba/src/runtime/virtcontainers/qemu.go#L1-L2

Reply:
Issue opened for addressing: 

https://github.com/kata-containers/kata-containers/issues/4251